site/content/entry/yuzu-progress-report-may-2022/index.md

```diff
@@ -0,0 +1,235 @@
+The issue only affects Polaris GPUs, and the extension is available to newer architectures too, like Vega or RDNA2 (we don’t talk about ~~Bruno~~ RDNA1).
+
+After a few debugging sessions we found out that yuzu’s implementation of VK_KHR_workgroup_memory_explicit_layout assumes that all GPUs compatible with it support 16-bit integer operations.
+While this was the case for all GPUs previous to AMDs implementation of the extension, Polaris is notorious for its lack of 16-bit precision support (shows its age, you could say), and as expected, forcing a GPU to do something it doesn’t support will result in a crash, hurray.
+
+toastUnlimited {{< gh-hovercard "8369" "disabled the extension" >}} on Polaris GPUs while we wait for our dedicated GPU devs to have the time to implement the proper fix, allow the extension to work with 32-bit precision.
+
+Now, still on the subject of AMD, Windows Vulkan drivers, and extension blocking.
+Since driver version 22.5.2, support was added for `VK_KHR_push_descriptor`, an old extension that has been working in every other driver for the past 5 years, be it Intel, NVIDIA or Mesa.
+
+We don’t know the cause, but only AMD’s Windows drivers crash when calling VK_KHR_push_descriptor, and since this extension is critical in the whole rendering process, any AMD GPU would crash on any game.
+
+Maybe this time we do have a broken implementation at release? It’s easy to believe so, this extension has been working since our first Vulkan release on any other driver.
+If that’s the case, it’s AMD’s turn to solve the issue.
+In the meantime, toastUnlimited {{< gh-hovercard "8379" "blocked the extension" >}} on AMD drivers with affected Vulkan driver versions.
+
+[asLody](https://github.com/asLody) {{< gh-hovercard "8311" "implemented stencil fixes when two faces are disabled." >}}
+This has the potential to improve rendering in native OpenGL games.
+
+## HLE Improvements
+
+In software engineering, a spinlock is a lock that causes a thread trying to acquire it to simply wait in a loop
+ ("spin") while repeatedly checking whether the lock is available.
+
+{{< imgs
+    "./spinlock.png| Example of a spinlock"
+  >}}
+
+On the other hand, the word "mutex" stands for an object providing `MUTual EXclusion` between threads. 
+Mutex ensures that only one thread has access to a critical section or data by using operations like a lock and unlock. 
+A thread that acquires the lock of mutex can use the critical section while other threads must wait till the lock is released.
+
+{{< imgs
+    "./mutex.png| Example of a mutex"
+  >}}
+
+In theory, when a thread tries to lock a mutex and it does not succeed, because the mutex is already locked, it will go to sleep, immediately allowing another thread to run. 
+It will continue to sleep until being woken up, which will be the case once the mutex is being unlocked by whatever thread was holding the lock before.
```

Comment:
The paused thread will continue to "sleep" until it is able to acquire the mutex. This may happen once the current thread holding the mutex lock releases it.